### PR TITLE
Refactor unneeded 'continue' jumps in models

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -2966,12 +2966,9 @@ def as_dict(self):
...
         )
         new_dict = dict(self.__dict__)
-        for key in new_dict:
-            if key in ["start_date", "end_date"]:
-                val = new_dict[key]
-                if not val or isinstance(val, str):
-                    continue
-                new_dict.update({key: val.isoformat()})
+        for key, val in new_dict.items():
+            if key in ["start_date", "end_date"] and isinstance(val, datetime):
```
Comment (potiuk) [inline airflow/models/taskinstance.py]: I think this is not exactly the same. We can have pendulum datetime here as well.

File: airflow/models/mappedoperator.py [hunk 2]
```diff
@@ -97,15 +97,11 @@ def validate_mapping_kwargs(op: type[BaseOperator], func: ValidationSource, valu
...
-            if func != "expand":
-                continue
-            if value is NOTSET:
-                continue
-            if is_mappable(value):
-                continue
-            type_name = type(value).__name__
-            error = f"{op.__name__}.expand() got an unexpected type {type_name!r} for keyword argument {name}"
-            raise ValueError(error)
+            if func == "expand" and value is not NOTSET and not is_mappable(value):
```
Comment (potiuk) [inline airflow/models/mappedoperator.py]: this condition has too many "not's" . I think previous version was more readable. I think. Maybe this ? :

```suggestion
            if func == "expand" and  not (value is NOTSET or  is_mappable(value)) :
```

or even extract it to separate function

Comment (eumiro) [reply airflow/models/mappedoperator.py]: Reduced the code further to more easily spot different exit points.

File: airflow/models/mappedoperator.py [hunk 3]
```diff
@@ -90,31 +90,21 @@ def validate_mapping_kwargs(op: type[BaseOperator], func: ValidationSource, valu
...
-            if func != "expand":
-                continue
-            if value is NOTSET:
-                continue
-            if is_mappable(value):
-                continue
-            type_name = type(value).__name__
-            error = f"{op.__name__}.expand() got an unexpected type {type_name!r} for keyword argument {name}"
-            raise ValueError(error)
+            if func == "expand" and not (value is NOTSET or is_mappable(value)):
```
Comment (potiuk) [inline airflow/models/mappedoperator.py:95]: Should we change `is_mappable` to return False when `value is NOTSET` ? That would change the condition to:

```
if func == "expand" and not is_mappable(value):
```

**Pull request closed without merging**